homeassistant/components/modbus/modbus.py

```diff
@@ -209,11 +209,12 @@ def close(self):
         """Disconnect client."""
         with self._lock:
             try:
-                self._client.close()
-                del self._client
-                self._client = None
+                if self._client:
+                    self._client.close()
+                    del self._client
```

Comment:
Is the `del` needed? It will be garbage collected by default if there are no references left.

Reply:
Python uses ref-counting, so when we set `self._client = None` at the line below, and it was the last reference, it will be garbage collected.
```suggestion
```

Reply:
Done. At least in older versions of Python the garbage collection was not immediate and we saw spurious problems (in another program) due to the fact that pymodbus was no deleted before being reopened.